log/field.go

```diff
@@ -0,0 +1,201 @@
+package log
+
+import "math"
+
+type fieldType int
+
+const (
+	stringType fieldType = iota
+	boolType
+	intType
+	int32Type
+	uint32Type
+	int64Type
+	uint64Type
+	float32Type
+	float64Type
+	errorType
+	objectType
+	lazyLoggerType
```

Comment:
You could also write something like:

LogFields("employee", struct{First, Last string}{first, last})

I experimented with nested logging--I ran into trouble with the semantics. Are labels tags? Can you apply more than one tag to a value? What's the resulting structure look like? In the end I did come up with a satisfying approach, but it was significantly more complicated than anything discussed in this thread, possibly beyond the point of actually being useful. This is why I ask--what the application is for nested logging.

Do you want to find all logs that have a "firstName" property? Is there any sort of namespace to prevent matching logs from unrelated code with "firstName" properties? Can you join logs by these keys?